src/librustc_mir/hair.rs

```diff
@@ -15,144 +15,78 @@
 //! structures.
 
 use repr::{BinOp, BorrowKind, Field, Literal, Mutability, UnOp};
-use std::fmt::Debug;
-use std::hash::Hash;
-
-pub trait Hair: Sized+Debug+Clone+Eq+Hash { // (*)
-
-    // (*) the `Sized` and Debug` bounds are the only ones that really
-    // make sense.  The rest are just there so that we can
-    // `#[derive(Clone)]` on things that are parameterized over
-    // `H:HAIR`. It's kind of lame.
```

Comment:
Since this trait is being removed, are the panic-only trait implementation for Cx still needed?
https://github.com/rust-lang/rust/pull/28866/files#diff-ec73b70c7997781e552e837799fe63bfL170